thoth/storages/graph/postgres.py

```diff
@@ -3851,9 +3852,17 @@ def create_python_cve_record(
         package_name = self.normalize_python_package_name(package_name)
         package_version = self.normalize_python_package_version(package_version)
         with self._session_scope() as session, session.begin(subtransactions=True):
-            cve, _ = CVE.get_or_create(
-                session, cve_id=record_id, version_range=version_range, advisory=advisory, cve_name=cve
-            )
+            cve = session.query(CVE).filter(CVE.cve_id == record_id).first()
```

Comment:
@fridex I just left `with_entities` out, so guessing that way all will request parameters like in SQL `*`.

Reply:
To optimize the query, you can use exists - that way the database will immediately return if matching result is found and will not return all the data (serialization/deserialization + data transfer overhead):

```python
session.query(exists().where(CVE.id==record_id))
```

It will also make sense to make `cve_id` not nullable and unique (indexed for better performance).

Reply:
Thanks for the advise. But looks like `cve = session.query(exists().where(CVE.id==record_id))` just constructs the query and if I add `scalar()` to the end of the query like `cve = session.query(exists().where(CVE.id==record_id)).scalar()`, then it returns a boolean whether the record exists or not.

Reply:
@fridex , should I stick to the first one which is `cve = session.query(CVE).filter(CVE.cve_id == record_id).first()` instead of `session.query(exists().where(CVE.id==record_id))`?

Reply:
Use the `exists` one, it has benefits stated above.

Reply:
@fridex I'm sorry, but I really want to understand. If we go with `exists` approach, then as I said it return True or False, right. Should I like, first find out whether that record exists, then execute another query to really get that record because we should return CVE object?
That's why for me it looks like 2 queries of which 1 is checking for existence and 2 is to really getting that data. However, I was suggesting instead to issue only one query where it tries to get the data immediately, so if we see that `None` is returned, then we can issue `CVE.get_or_create()` which will create that data, otherwise if something returned, we will return it?

Reply:
The most used scenario is when we already have the given CVE record (as we run this component CVE update job as a cron job for the same set of CVEs, that can get updated over time). So in most cases, we will hit "CVE is already present in the database" - that's why I'm suggesting that simple `exists` check that has minimal overhead.

If we do not have the given CVE with `cve_id` in the database, we can create it (not with `get_or_create` as we already know its not in the database).